### PR TITLE
Better flight SQL example codes

### PR DESCRIPTION
**Pull request by sundy-li**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 
Provide more detail examples and test codes for flight SQL.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by sundy-li** [comment]
I profiled the test case `test_select_1` and found that prepare cause much more latency in TcpClient.

```
 cargo test --release --package arrow-flight --example flight_sql_server --all-features --features flight-sql-experimental --features tls -- tests::test_select_1 --exact --nocapture 

running 1 test
testing uds client
prepare cost: 761.35µs
execute cost: 671.36µs
do get cost: 716.169µs
Client finished!
=======
testing tcp client
prepare cost: 41.503622ms
execute cost: 1.55827ms
do get cost: 1.58378ms
Client finished!
=======
testing https client
prepare cost: 1.39358ms
execute cost: 1.31692ms
do get cost: 1.354041ms
Client finished!
=======
test tests::test_select_1 ... ok
```

**Comment by sundy-li** [comment]
> no ideas about the reason.

Oh, it's due to TcpNoDelay, it's not enabled.

File: arrow-flight/Cargo.toml [hunk 1]
```diff
@@ -34,17 +34,37 @@ arrow-cast = { workspace = true }
 arrow-ipc = { workspace = true }
 arrow-schema = { workspace = true }
 base64 = { version = "0.21", default-features = false, features = ["std"] }
-tonic = { version = "0.9", default-features = false, features = ["transport", "codegen", "prost"] }
+tonic = { version = "0.9", default-features = false, features = [
```
Comment (tustvold) [inline arrow-flight/Cargo.toml:37]: I personally find this formatting harder to read...

**Review by tustvold — APPROVED** [review]
Looks good to me, mostly just some minor nits. I took the liberty of creating #4145 which should remove the need for the once_cell shenanigans. Happy to update this PR once that is in.

**Comment by tustvold** [comment]
Working on updating this

**Comment by tustvold** [comment]
Thank you for this :+1:

**Pull request merged**